### PR TITLE
Tpacket v3 6

### PR DESCRIPTION
**Pull request by regit**
Address comments made on #2015.

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/162
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/158

### CODE REVIEW

File: src/source-af-packet.c [hunk 1]
```diff
@@ -1388,14 +1569,188 @@ frame size: TPACKET_ALIGN(snaplen + TPACKET_ALIGN(TPACKET_ALIGN(tp_hdrlen) + siz
...
 }
 
+#ifdef HAVE_TPACKET_V3
+static int AFPComputeRingParamsV3(AFPThreadVars *ptv)
+{
+    ptv->req3.tp_block_size = ptv->block_size;
+    ptv->req3.tp_frame_size = 2048;
+    int frames_per_block = ptv->req3.tp_block_size / ptv->req3.tp_frame_size;
+    if (frames_per_block == 0) {
+        SCLogInfo("frame size to big");
```
Comment (inliniac) [inline src/source-af-packet.c]: This should be an error/warning? Also, it seems odd to state the frame size is wrong while it's hardcoded. Proper warning should say blocksize too small?

File: src/source-af-packet.c [hunk 2]
```diff
@@ -892,6 +907,135 @@ int AFPReadFromRing(AFPThreadVars *ptv)
...
+}
+
+static inline int AFPWalkBlock(AFPThreadVars *ptv, struct tpacket_block_desc *pbd, const int block_num)
+{
+    int num_pkts = pbd->hdr.bh1.num_pkts, i;
+    uint8_t *ppd;
+
+    ppd = (uint8_t *)pbd + pbd->hdr.bh1.offset_to_first_pkt;
+    for (i = 0; i < num_pkts; ++i) {
+        AFPParsePacketV3(ptv, pbd, (struct tpacket3_hdr *)ppd);
```
Comment (inliniac) [inline src/source-af-packet.c]: errors are ignored here

File: src/source-af-packet.c [hunk 3]
```diff
@@ -892,6 +907,135 @@ int AFPReadFromRing(AFPThreadVars *ptv)
...
+    ptv->pkts++;
+    p->livedev = ptv->livedev;
+    p->datalink = ptv->datalink;
+
+    if (ptv->flags & AFP_ZERO_COPY) {
+        if (PacketSetData(p, (unsigned char*)ppd + ppd->tp_mac, ppd->tp_snaplen) == -1) {
+            TmqhOutputPacketpool(ptv->tv, p);
+            SCReturnInt(AFP_FAILURE);
+        } else {
+            p->afp_v.relptr = pbd;
```
Comment (inliniac) [inline src/source-af-packet.c]: if condition exits the function, so this doesn't need to be in 'else' condition.

File: src/source-af-packet.c [hunk 4]
```diff
@@ -1388,14 +1569,188 @@ frame size: TPACKET_ALIGN(snaplen + TPACKET_ALIGN(TPACKET_ALIGN(tp_hdrlen) + siz
...
+static int AFPComputeRingParamsV3(AFPThreadVars *ptv)
+{
+    ptv->req3.tp_block_size = ptv->block_size;
+    ptv->req3.tp_frame_size = 2048;
+    int frames_per_block = ptv->req3.tp_block_size / ptv->req3.tp_frame_size;
+    if (frames_per_block == 0) {
+        SCLogInfo("frame size to big");
+        return -1;
+    }
+    ptv->req3.tp_frame_nr = ptv->ring_size;
```
Comment (inliniac) [inline src/source-af-packet.c]: tp_frame_nr is set here, but set again a few lines below

File: src/source-af-packet.c (hunk 1 above)
Comment (regit) [inline src/source-af-packet.c]: Right.

File: src/source-af-packet.c (hunk 4 above)
Comment (regit) [inline src/source-af-packet.c]: Oups.

File: src/source-af-packet.c [hunk 5]
```diff
@@ -892,6 +907,135 @@ int AFPReadFromRing(AFPThreadVars *ptv)
...
+    if (TmThreadsSlotProcessPkt(ptv->tv, ptv->slot, p) != TM_ECODE_OK) {
+        AFPFlushBlock(pbd);
+        TmqhOutputPacketpool(ptv->tv, p);
+        SCReturnInt(AFP_FAILURE);
+    }
+
+    SCReturnInt(AFP_READ_OK);
+}
+
+static inline int AFPWalkBlock(AFPThreadVars *ptv, struct tpacket_block_desc *pbd, const int block_num)
```
Comment (inliniac) [inline src/source-af-packet.c]: block_num (which is passed to this func as ptv->frame_offset), seems unused

File: src/source-af-packet.c [hunk 6]
```diff
@@ -892,6 +907,135 @@ int AFPReadFromRing(AFPThreadVars *ptv)
...
+{
+    int num_pkts = pbd->hdr.bh1.num_pkts, i;
+    uint8_t *ppd;
```
Comment (inliniac) [inline src/source-af-packet.c]: what does ppd mean?

File: src/source-af-packet.c (hunk 4 above)
Comment (regit) [inline src/source-af-packet.c]: done.

File: src/source-af-packet.c (hunk 2 above)
Comment (regit) [inline src/source-af-packet.c]: done.

File: src/source-af-packet.c (hunk 6 above)
Comment (regit) [inline src/source-af-packet.c]: I've reused variable name from function example. Think it could be Pointer to Packet Data.

**Pull request closed without merging**